### PR TITLE
Added collisions and scaling

### PR DESCRIPTION
**Pull request by theoden8**
Hey! I have added collisions! Push my code to the master and we will remain friends !

### CODE REVIEW

File: src/physical_object.py [hunk 1]
```diff
@@ -33,16 +33,39 @@ def get_forces_composition(self):
...
+        ret = [
+            (u1[0] * (m1 - m2) + 2 * m2 * u2[0]) / (m1 + m2),
+            (u1[1] * (m1 - m2) + 2 * m2 * u2[1]) / (m1 + m2)
+        ]
+        return [ret[0] - s.speed[0], ret[1] - s.speed[1]]
+
     def tick(self):
         """Tick the physical object: update its physical properties."""
+        s = self
         # print(str(self))
```
Comment (PolarBearSaviour) [inline src/physical_object.py]: This can be removed

Comment (mallok-s) [inline src/physical_object.py]: Can we remove commented out code please

File: src/physical_object.py [hunk 2]
```diff
@@ -33,16 +33,39 @@ def get_forces_composition(self):
...
 
+    def collide(self, q):
+        """
+        Dirtily calculates collisions.
+
+        :q: other object
+
+        :returns: new speed
+        """
+        s = self
```
Comment (mallok-s) [inline src/physical_object.py]: Shouldn't we being nouns as variable names? Might help improve the readability

**Review by mallok-s — COMMENTED** [review]
The code seems quite clean. The s =self is a clever way of making the code shorter. Possible could use more detailed doc strings or comments. No idea if it works, so I don't think I can approve for merge (sorry, hope we'll remain friends!)

**Comment by theoden8** [comment]
@PolarBearSaviour, now it does work and very well. Please, reconsider: it is so lovely now! This is the feature to be proud of. 🎱 And yes, you are, of course, welcome to review it. Although it is monkey code, it is fast to write and easy to understand.

### Summary:

* The objects collide with full elasticity (complexity: N^2).
* A bit of mass of the smaller object transfers to the bigger object.
* When an object becomes very small, it is removed from the model as it is not displayable anyway, but causes laggs.
* Possible extension (doesnt work yet) of spawning asteroids after collision instead of transfering masses.

File: src/physical_object.py (hunk 2 above)
Comment (theoden8) [reply src/physical_object.py]: As long as it is clear what it takes, shouldn't matter. Although if I used p and q to represent different things in different places, that would make perfect sense.

File: src/physical_object.py (hunk 1 above)
Comment (theoden8) [reply src/physical_object.py]: The reason to keep it there is to make debugging faster: we have all the debugging functions in correct places, whenever something goes wrong, we just uncomment them and faster navigate to the right place.

**Review by mallok-s — COMMENTED** [review]
Kirill do you know if there issues with pygame and windows? It seems to work on mac (obviously) but on windows the code does nothing, like the screen just opens up and then nothing moves

**Comment by theoden8** [comment]
Interesting. PyGame is a python wrapper for SDL2, which is cross platform. If the computer is not very fast, increase the gravitational constant by a power of ten, or decrease the number of elements in the model. Either way, you should be able to see how fast does the word "tick" appear.

**Review by mallok-s — APPROVED** [review]
The code is fine and I can get it working on some windows machines. Potentially the code could contain more commentary but seems reasonable readable. Though I don't understand why we are using external libraries for physical constants, can we just ditch scripy and code in the constants ourselves?

**Comment by theoden8** [comment]
The reason to use external libraries for constants is because there might be something useful like light years, all in one. I just can't be bothered finding them on my own.

**Pull request merged**